File: src/main/java/com/zunza/buythedip/auth/oauth2/dto/GoogleOAuth2Response.java
package com.zunza.buythedip.auth.oauth2.dto;

import java.util.Map;

import com.zunza.buythedip.user.constant.OAuth2Provider;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class GoogleOAuth2Response implements OAuth2Response {
	private final Map<String, Object> attribute;

	@Override
	public OAuth2Provider getProvider() {
		return OAuth2Provider.GOOGLE;
	}

	@Override
	public String getProviderId() {
		return attribute.get("sub").toString();
	}

	@Override
	public String getEmail() {
		return attribute.get("email").toString();
	}
}

File: src/main/kotlin/com/zunza/buythedip/auth/oauth2/dto/GoogleOAuth2Response.kt
package com.zunza.buythedip.auth.oauth2.dto

import com.zunza.buythedip.user.constant.OAuth2Provider

class GoogleOAuth2Response(
    private val attribute: Map<String, Any>
) : OAuth2Response {

    override fun getProvider(): OAuth2Provider {
        return OAuth2Provider.GOOGLE
    }

    override fun getProviderId(): String {
        return attribute["sub"].toString()
    }

    override fun getEmail(): String {
        return attribute["email"].toString()
    }
}
